pkgs/development/python-modules/pynut2/default.nix

```diff
@@ -0,0 +1,36 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, requests
+#, pytestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "pynut2";
+  version = "2.1.2";
+
+  src = fetchFromGitHub {
+    owner = "mezz64";
+    repo = "python-nut2";
+    rev = "${version}";
+    sha256 = "1lg7n1frndfgw73s0ssl1h7kc6zxm7fpiwlc6v6d60kxzaj1dphx";
+  };
+
+  propagatedBuildInputs = [
+    requests
+  ];
+
+  # tests are broken, they import 'nut2', not 'pynut2.nut2'
+  #checkInputs = [
```

Comment:
Something like this could help to get the tests running:

```bash
  postPatch = ''
    substituteInPlace tests/test_nutclient.py \
      --replace "from nut2 import PyNUTClient" "from pynut2.nut2 import PyNUTClient"
  '';
```

Reply:
Thank you, but after a more in depth look I noticed it's a lost cause

The tests are for and old api of the lib. I tried playing with those a bit, but I am not a python programmer and it's well beyond a few substitutions